src/daemon/mod.rs

```diff
@@ -506,9 +505,9 @@ async fn fetch_snapshot_if_required(
             })
             .await;
             if !have_permission {
-                bail!("Forest requires a snapshot to sync with the network, but automatic fetching is disabled.")
+                std::process::exit(0);
```

Comment:
Why?

Reply:
Reverted. This was a temp change to fix daemon e2e tests, moving `exit_after_init` check to before snapshot import instead